app/models/account.rb

```diff
@@ -4,31 +4,30 @@
 class Account < ApplicationRecord
   has_many :users, dependent: :nullify
   # rubocop:disable Rails/HasManyOrHasOneDependent
-  has_many :hosts, foreign_key: :account, primary_key: :account_number,
+  has_many :hosts, foreign_key: :org_id, primary_key: :org_id,
                    inverse_of: :account_object
   # rubocop:enable Rails/HasManyOrHasOneDependent
   has_many :profiles, dependent: :destroy
   has_many :policies, dependent: :destroy
   has_many :business_objectives, through: :policies
 
-  validates :account_number, presence: true, allow_blank: false
+  validates :org_id, presence: true, allow_blank: false
 
   attr_accessor :identity_header
 
   class << self
     def from_identity_header(identity_header)
-      # FIXME: swap the 'account_number' with the 'org_id' after the translation is done
       account = Account.find_or_create_by(
-        account_number: identity_header.identity['account_number']
+        org_id: identity_header.identity['org_id']
```

Comment:
When testing locally, I'm getting the `org_id` inside of `internal` hash.
Checking if it should be changed or if it's just in my local setup:

```diff
diff --git a/app/models/account.rb b/app/models/account.rb
index ac3a238c..91778df5 100644
--- a/app/models/account.rb
+++ b/app/models/account.rb
@@ -18,9 +18,15 @@ class Account < ApplicationRecord
   class << self
     def from_identity_header(identity_header)
       account = Account.find_or_create_by(
-        org_id: identity_header.identity['org_id']
+        org_id: identity_header.identity["internal"]['org_id']
       )
```

Reply:
Should work as it is now.